modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbStorageEngine.java

```diff
@@ -165,4 +170,17 @@ public RocksDbTableStorage createMvTable(TableConfiguration tableCfg) throws Sto
 
         return new RocksDbTableStorage(this, tablePath, tableCfg, dataRegion);
     }
+
+    /** {@inheritDoc} */
+    @Override public TxnStateTableStorage createTxnStateTableStorage(TableConfiguration tableCfg) {
+        Path path = storagePath.resolve(TABLE_DIR_PREFIX + tableCfg.value().tableId()).resolve(TXN_STATE_DIR);
+
+        try {
+            Files.createDirectories(path);
+        } catch (IOException e) {
+            throw new StorageException("Failed to create transaction state storage directory for " + tableCfg.value().name(), e);
+        }
+
+        return new TxnStateRocksDbTableStorage(tableCfg, path, scheduledPool, threadPool, configuration().value()::flushDelayMillis);
```

Comment:
Should we have a separate configuration for this storage? I don't remember this part ever being discussed.
Maybe we should discuss it actually.